pkg/service/FluxService/FluxApplicationService.go

```diff
@@ -0,0 +1,420 @@
+		req := &client.ExternalResourceTreeRequest{
+			ClusterConfig:          request.Config,
+			ExternalResourceDetail: fluxK8sResourceList,
+		}
+		resourceTreeResponse, err := impl.common.GetResourceTreeForExternalResources(req)
+		if err != nil {
+			return nil, appStatus, fmt.Errorf("failed to get resource tree for external resources: %v", err)
+		}
+		if len(resourceTreeResponse.Nodes) > 0 {
+			fluxAppTreeResponse = append(fluxAppTreeResponse, resourceTreeResponse)
+		}
+	}
+	return fluxAppTreeResponse, appStatus, err
+}
+func (impl *FluxApplicationServiceImpl) getTreeForHrList(request *FluxAppDetailRequest, fluxHrList []*FluxHr) ([]*bean.ResourceTreeResponse, error) {
+	fluxAppTreeResponse := make([]*bean.ResourceTreeResponse, 0)
+	var err error
+	for _, fluxHr := range fluxHrList {
+		releaseName := fluxHr.Name
+		namespace := fluxHr.Namespace
+		helmRelease, err := impl.common.GetHelmRelease(request.Config, namespace, releaseName)
+		if err != nil {
+			impl.logger.Errorw("error in getting helm release", "err", err)
+		}
+		appDetailRequest := &client.AppDetailRequest{
+			ReleaseName:   releaseName,
+			Namespace:     namespace,
+			ClusterConfig: request.Config,
+		}
+		if helmRelease != nil {
+			resourceTreeResponse, err := impl.common.BuildResourceTree(appDetailRequest, helmRelease)
+			if err != nil {
+				//return nil, appStatus, fmt.Errorf("failed to build resource tree for Helm release: %v", err)
+				impl.logger.Errorw("error in building resource tree for HelmResource", "err", err)
+			}
+			fluxAppTreeResponse = append(fluxAppTreeResponse, resourceTreeResponse)
+		}
+
+	}
+	return fluxAppTreeResponse, err
+}
+func (impl *FluxApplicationServiceImpl) getFluxResourceAndFluxHrList(app *FluxAppDetailRequest, fluxK8sResourceList *[]*client.ExternalResourceDetail, fluxHrList *[]*FluxHr, obj map[string]interface{}) {
+	childResourcesList, err := fetchInventoryList(obj)
+	if err != nil {
+		impl.logger.Errorw("error in getting inventory list", "err", err)
+		return
+	}
+	isKubeConfig, err := getFluxSpecKubeConfig(obj)
+	if err != nil {
+		impl.logger.Errorw("error in getting flux spec kubeConfig", "err", err)
+		return
+	}
+	for _, childResource := range childResourcesList {
+		*fluxK8sResourceList = append(*fluxK8sResourceList, &client.ExternalResourceDetail{
+			Namespace: childResource.Namespace,
+			Name:      childResource.Name,
+			Group:     childResource.Group,
+			Kind:      childResource.Kind,
+			Version:   childResource.Version,
+		})
+		if childResource.Group == FluxKustomizationGroup && childResource.Kind == FluxAppKustomizationKind && childResource.Name == app.Name && childResource.Namespace == app.Namespace {
+			continue
+		}
+		if childResource.Group == FluxHelmReleaseGroup &&
+			childResource.Kind == FluxAppHelmreleaseKind {
+			var releaseName, nameSpace string
+			releaseName, nameSpace, _, err = impl.getHelmReleaseNameNamespaceandStatusDetail(childResource.Name, childResource.Namespace, app.Config)
```

Comment:
getHelmReleaseNameNamespaceandStatusDetail this func has only one implementation as we are eating away FluxAppStatusDetail (the 3rd resp) struct, so pls remove that